src/zenml/zen_server/deploy/helm/templates/hpa.yaml

```diff
@@ -17,12 +17,16 @@ spec:
     - type: Resource
       resource:
         name: cpu
-        targetAverageUtilization: {{ .Values.autoscaling.targetCPUUtilizationPercentage }}
+        target:
+          type:  Utilization
+          averageUtilization: {{ .Values.autoscaling.targetCPUUtilizationPercentage }}
```

Comment:
Same question here: is this change required because some Kubernetes cluster APIs don't support the previous syntax ? If so, shouldn't it be in an if/else condition to maintain backwards compatibility with previous versions ?

Reply:
Yes, the new syntax is need for v2 of the HPA resource. The older versions would no longer be supported. What I can do is add a different chart version that has support for v1beta1 for the same zenml version but that might need some changes across the board. Still the safest bet and allows us to support everyone.

Reply:
You don't need a different chart version, just an if condition here. See this thread for example: https://stackoverflow.com/questions/70346448/deploy-conditional-charts-based-on-kubeversion-helm